src/app/pages/fun-zone/simon/leaderbord/leaderbord.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Firestore, doc } from "@angular/fire/firestore"
+import { Component, OnInit, ViewChild } from "@angular/core"
+import { getDoc } from "firebase/firestore"
+import { __values } from "tslib"
+import { MatTable } from "@angular/material/table"
+
+export interface RowElement {
+  position: number
+  name: string
+  score: number
+}
+
+const ELEMENT_DATA: RowElement[] = []
+
+@Component({
+  selector: "app-leaderbord",
+  templateUrl: "./leaderbord.component.html",
+  styleUrls: ["./leaderbord.component.scss"],
+})
+export class LeaderbordComponent implements OnInit {
+  dataSource = [...ELEMENT_DATA]
+  highScores = new Map()
+  names: [string]
+  displayedColumns: string[] = ["position", "name", "score"]
+  constructor(private firestore: Firestore) {
+    this.names = [""]
+  }
+
+  @ViewChild(MatTable) table!: MatTable<RowElement>
+
+  ngOnInit(): void {
+    this.updateTable()
+  }
+  async getHighScoresArray(): Promise<Map<string, number>> {
+    const document = await getDoc(doc(this.firestore, "high scores/simon"))
```

Comment:
It might be better to use onSnapshot to update the high scores, instead of the user having to click a button every time